### PR TITLE
fix(ledger): skip rollback if it is ahead of ledger tip

### PR DESCRIPTION
**Pull request by wolf31o2**
## Summary by cubic
Skip metadata rollback when the requested rollback point is ahead of the `ledger` tip. Prevents advancing a lagging ledger during catch-up and avoids unnecessary resyncs.

- **Bug Fixes**
  - Guarded `rollback` to no-op (with debug log) if rollback point slot > current tip slot.
  - Added `TestTryResolveForkDoesNotAdvanceLaggingLedgerTip` to verify the ledger tip and DB tip remain unchanged and no resync event is emitted when the raw chain is ahead.

<sup>Written for commit d753253f4820f6e1e2aa1598251e481d80c15d04. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for chain synchronization fork resolution when the ledger tip lags behind the blockchain state.

* **Bug Fixes**
  * Enhanced rollback mechanism to properly handle edge cases where the requested rollback point is ahead of the current ledger state, preventing unintended metadata rollbacks.

[Review Change Stack](https://app.coderabbit.ai/change-stack/blinklabs-io/dingo/pull/2242)

### CODE REVIEW

File: ledger/chainsync_rollback_test.go [hunk 1]
```diff
@@ -278,6 +278,109 @@ func TestTryResolveForkSynchronizesLedgerTip(t *testing.T) {
...
+		fixture.ancestorNonce...,
+	)
+	preRollbackSeq := fixture.ls.lastLocalRollbackSeq
+
+	forkHash := testHashBytes("ahead-raw-chain-fork")
+	header := mockHeader{
+		hash:        lcommon.NewBlake2b256(forkHash),
+		prevHash:    lcommon.NewBlake2b256(fixture.currentTip.Point.Hash),
+		blockNumber: fixture.currentTip.BlockNumber + 2,
+		slot:        fixture.currentTip.Point.Slot + 20,
```
[the remark this answers is not in the document]
Comment (wolf31o2) [reply ledger/chainsync_rollback_test.go:337]: Did you hallucinate this problem?

**Review by musik-c — APPROVED** [review]
